Makefile

```diff
@@ -136,7 +136,9 @@ ifdef CONFIG_HTTP
   http_LIBS = -lcurl -lssl -lcrypto
   ENGINES += http
 endif
-SOURCE += oslib/asprintf.c
+ifeq "$(or $(CONFIG_HAVE_ASPRINTF),$(CONFIG_HAVE_VASPRINTF))" ""
```

Comment:
I'm not sure this is an improvement. Always building asprintf.c helps to catch build errors in the implementation even when developing on a platform (Linux) that provides an asprintf() implementation.

Reply:
Fair enough (I think I was just trying to speed up Windows builds which seem to be painfully slow).

Reply:
What kind of environment are you using to build fio for Windows? On a five year old computer it takes less than 20 seconds to configure and cross-compile fio:
~~~
$ cat fio-rebuild-with-mingw
#!/bin/sh

git clean -f -d >/dev/null
./configure --cc=x86_64-w64-mingw32-gcc --extra-cflags='-Wall -Wextra -Wno-unused-parameter -Wno-sign-compare -Wno-implicit-fallthrough -Werror' --disable-optimizations
if which gmake >/dev/null 2>&1; then
    MAKE=gmake
else
    MAKE=make
fi
$MAKE -j$(nproc)
$ fio-rebuild-with-mingw
Forcing some known good options on Windows
Operating system              CYGWIN
CPU                           x86_64
Big endian                    no
Target Windows version        7
Compiler                      x86_64-w64-mingw32-gcc
Cross compile                 yes

Static build                  yes
Wordsize                      32
zlib                          no
Linux AIO support             no
Linux AIO support rw flags    no
Linux AIO over io_uring       no
POSIX AIO support             no
POSIX AIO support needs -lrt  no
POSIX AIO fsync               no
POSIX pshared support         no
pthread_condattr_setclock()   no
pthread_sigmask()             yes
Solaris AIO support           no
__sync_fetch_and_add          yes
__sync_synchronize            yes
__sync_val_compare_and_swap   yes
libverbs                      no
rdmacm                        no
asprintf()                    yes
vasprintf()                   no
Linux fallocate               no
POSIX fadvise                 no
POSIX fallocate               no
sched_setaffinity(3 arg)      no
sched_setaffinity(2 arg)      no
clock_gettime                 yes
CLOCK_MONOTONIC               yes
CLOCK_MONOTONIC_RAW           no
CLOCK_MONOTONIC_PRECISE       no
clockid_t                     yes
gettimeofday                  yes
fdatasync                     yes
pipe()                        no
pipe2()                       no
pread()                       no
sync_file_range               no
EXT4 move extent              no
Linux splice(2)               no
GUASI                         no
libnuma                       no
strsep                        no
strcasestr                    no
strlcat                       no
getopt_long_only()            yes
inet_aton                     no
socklen_t                     yes
__thread                      yes
RUSAGE_THREAD                 yes
SCHED_IDLE                    yes
TCP_NODELAY                   yes
Net engine window_size        yes
TCP_MAXSEG                    no
RLIMIT_MEMLOCK                no
pwritev/preadv                no
pwritev2/preadv2              no
IPv6 helpers                  no
http engine                   no
Rados engine                  no
Rados Block Device engine     no
setvbuf                       yes
Gluster API engine            no
s390_z196_facilities          no
HDFS engine                   no
MTD                           no
libpmem                       no
libpmem1_5                    no
libpmemblk                    no
PMDK pmemblk engine           no
PMDK dev-dax engine           no
PMDK libpmem engine           no
DDN's Infinite Memory Engine  no
iscsi engine                  no
NBD engine                    no
lex/yacc for arithmetic       no
getmntent                     no
getmntinfo                    no
Static Assert                 yes
bool                          yes
strndup                       no
Valgrind headers              no
libzbc engine                 no
march_armv8_a_crc_crypto      no
cuda                          no
Build march=native            no
CUnit                         no
__kernel_rwf_t                no
-Wimplicit-fallthrough=2      yes
MADV_HUGEPAGE                 no
gettid                        no
statx(2)/libc                 no
statx(2)/syscall              no
Consider installing zlib-dev (zlib-devel, some fio features depend on it.
Note that some distros have separate packages for static libraries.
Lib-based ioengines dynamic   no
TCMalloc support              no
FIO_VERSION = fio-3.22-45-g8c8a-dirty
    CC crc/crc16.o
    CC crc/crc32.o
    CC crc/crc32c-arm64.o
    CC crc/crc32c-intel.o
    CC crc/crc32c.o
    CC crc/crc64.o
    CC crc/crc7.o
    CC crc/fnv.o
    CC crc/md5.o
    CC crc/murmur3.o
    CC crc/sha1.o
    CC crc/sha256.o
    CC crc/sha3.o
    CC crc/sha512.o
    CC crc/test.o
    CC crc/xxhash.o
    CC lib/axmap.o
    CC lib/bloom.o
    CC lib/flist_sort.o
    CC lib/gauss.o
    CC lib/getrusage.o
    CC lib/hweight.o
    CC lib/ieee754.o
    CC lib/lfsr.o
    CC lib/memalign.o
    CC lib/memcpy.o
    CC lib/mountcheck.o
    CC lib/num2str.o
    CC lib/output_buffer.o
    CC lib/pattern.o
    CC lib/prio_tree.o
    CC lib/rand.o
    CC lib/rbtree.o
    CC lib/strntol.o
    CC lib/zipf.o
    CC gettime.o
    CC ioengines.o
    CC init.o
    CC stat.o
    CC log.o
    CC time.o
    CC filesetup.o
    CC eta.o
    CC verify.o
    CC memory.o
    CC io_u.o
    CC parse.o
    CC fio_sem.o
    CC rwlock.o
    CC pshared.o
    CC options.o
    CC smalloc.o
    CC filehash.o
    CC profile.o
    CC debug.o
    CC engines/cpu.o
    CC engines/mmap.o
    CC engines/sync.o
    CC engines/null.o
    CC engines/net.o
    CC engines/ftruncate.o
    CC engines/filecreate.o
    CC engines/filestat.o
    CC server.o
    CC client.o
    CC iolog.o
    CC backend.o
    CC libfio.o
    CC flow.o
    CC cconv.o
    CC gettime-thread.o
    CC helpers.o
    CC json.o
    CC idletime.o
    CC td_error.o
    CC profiles/tiobench.o
    CC profiles/act.o
    CC io_u_queue.o
    CC filelock.o
    CC workqueue.o
    CC rate-submit.o
    CC optgroup.o
    CC helper_thread.o
    CC steadystate.o
    CC zone-dist.o
    CC zbd.o
    CC engines/windowsaio.o
    CC oslib/asprintf.o
    CC oslib/strsep.o
    CC oslib/strcasestr.o
    CC oslib/strlcat.o
    CC oslib/strndup.o
    CC oslib/inet_aton.o
    CC oslib/statx.o
    CC os/windows/cpu-affinity.o
    CC os/windows/posix.o
    CC fio.o
    CC t/log.o
    CC t/genzipf.o
    CC t/dedupe.o
    CC t/debug.o
    CC t/arch.o
    CC t/verify-state.o
    CC t/stest.o
    CC t/ieee754.o
    CC t/axmap.o
    CC t/lfsr-test.o
    CC t/gen-rand.o
    CC t/memlock.o
  LINK t/fio-genzipf
  LINK t/ieee754
  LINK t/axmap
  LINK t/gen-rand
  LINK t/memlock
  LINK t/stest
  LINK t/lfsr-test
  LINK fio
  LINK t/fio-dedupe
  LINK t/fio-verify-state

real	0m19.733s
user	1m46.673s
sys	0m8.462s

Compilation finished at Sun Sep  6 12:03:36
~~~